libnetwork/networkdb/networkdb_test.go

```diff
@@ -428,6 +428,22 @@ func TestNetworkDBCRUDMediumCluster(t *testing.T) {
 
 	dbs := createNetworkDBInstances(t, n, "node", DefaultConfig())
 
+	// Shake out any data races.
+	done := make(chan struct{})
+	defer close(done)
+	for _, db := range dbs {
+		go func(db *NetworkDB) {
+			for {
+				select {
+				case <-done:
+					return
+				default:
+				}
+				_ = db.GetTableByNetwork("test_table", "network1")
+			}
+		}(db)
+	}
```

Comment:
Definitely non-blocking; guess we could make this use a context to make it slight more to the point

```go
// Shake out any data races.
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
for _, db := range dbs {
	go func(db *NetworkDB) {
		for ctx.Err() == nil {
			_ = db.GetTableByNetwork("test_table", "network1")
		}
	}(db)
}
```

Reply:
Fair point. Also https://pkg.go.dev/testing#T.Context

Reply:
Ha! I was just looking at that one earlier Today, but we need to wait for go1.24 (which should be possible now that 1.24.3 is released). I was looking for it to be used in our integration tests, but we have a `setupTest` that depends on the `TestMain` to set up a common context, which doesn't allow creating one (`testing.M` doesn't have the `Context()` 😂